apps/web/playwright/insights.e2e.ts

```diff
@@ -0,0 +1,276 @@
+    expect(page.url()).toContain("teamId=");
+
+    // get div from team select filter with this class flex flex-col gap-0.5 [&>*:first-child]:mt-1 [&>*:last-child]:mb-1
+    await page.getByTestId("dashboard-shell").getByText("Team: test-insights").click();
+    await page
+      .locator('div[class="flex flex-col gap-0.5 [&>*:first-child]:mt-1 [&>*:last-child]:mb-1"]')
+      .click();
+    const teamSelectFilter = await page.locator(
+      'div[class="hover:bg-muted flex items-center py-2 pl-3 pr-2.5 hover:cursor-pointer"]'
+    );
+
+    await expect(teamSelectFilter).toHaveCount(3);
+
+    // switch to self profile
+    await page.getByTestId("dashboard-shell").getByText("Your Account").click();
+    // wait for url to change and have userId=
+    await page.waitForURL((url) => {
+      const userId = url.searchParams.get("userId");
+      return !!userId;
+    });
+
+    expect(page.url()).toContain("isAll=false");
+    expect(page.url()).toContain("userId=");
+
+    // switch to team 1
+    await page.getByTestId("dashboard-shell").getByText("test-insights").nth(0).click();
+    await page.waitForLoadState("networkidle");
+
+    expect(page.url()).toContain("isAll=false");
+    expect(page.url()).toContain("teamId=");
+
+    // switch to team 2
+    await page.getByTestId("dashboard-shell").getByText("test-insights-2").click();
+    await page.waitForLoadState("networkidle");
+    expect(page.url()).toContain("isAll=false");
+    expect(page.url()).toContain("teamId=");
+  });
+
+  test("should be able to switch between memberUsers", async ({ page, users }) => {
+    const owner = await users.create();
+    const member = await users.create();
+
+    await createTeamsAndMembership(owner.id, member.id);
+
+    await owner.apiLogin();
+
+    await page.goto("/insights");
+    await page.waitForLoadState("networkidle");
+
+    // expect to have isAll=true and teamId= in query params
+    expect(page.url()).toContain("isAll=false");
+    expect(page.url()).toContain("teamId=");
+
+    await page.getByText("Add filter").click();
+
+    await page.getByRole("button", { name: "User" }).click();
+    // <div class="flex select-none truncate font-medium" data-state="closed">People</div>
+    await page.locator('div[class="flex select-none truncate font-medium"]').getByText("People").click();
+
+    await page
+      .locator('div[class="hover:bg-muted flex items-center py-2 pl-3 pr-2.5 hover:cursor-pointer"]')
+      .nth(0)
+      .click();
+    await page.waitForLoadState("networkidle");
+
+    await page
+      .locator('div[class="hover:bg-muted flex items-center py-2 pl-3 pr-2.5 hover:cursor-pointer"]')
+      .nth(1)
+      .click();
+    await page.waitForLoadState("networkidle");
+
+    // press escape button to close the filter
+    await page.keyboard.press("Escape");
+
+    await page.getByRole("button", { name: "Clear" }).click();
+
+    expect(page.url()).not.toContain("memberUserId=");
```

Comment:
@alannnc tests are failing probably because it takes some time for the url to update after clicking "Clear"